core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AzureBatchSource.java

```diff
@@ -32,26 +32,28 @@
 @Name("AzureBlobStore")
 @Description("Batch source to read from Azure Blob Storage.")
 public class AzureBatchSource extends FileBatchSource {
-
   @SuppressWarnings("unused")
   private final AzureBatchConfig config;
 
   public AzureBatchSource(AzureBatchConfig config) {
-    super(config);
+    super(new FileBatchConfig(config.referenceName, config.fileRegex, config.timeTable, config.inputFormatClass,
```

Comment:
hm, I see. Can you do another refactor to change the current FileBatchSource to:

```
abstract class AbstractFileBatchSource<T extends FileSourceConfig> extends ReferenceBatchSource<Object, Object, StructuredRecord>
```

Then FileBatchSource, AzureBatchSource, and S3BatchSource all extend from the abstract class